cogs/strike.py

```diff
@@ -698,6 +699,15 @@ async def on_member_update(self, before: discord.Member, after: discord.Member)
         if not after.timed_out or before.timed_out == after.timed_out:
             return
 
+        audit_log_entry: discord.AuditLogEntry
+        async for audit_log_entry in main_guild.audit_logs(limit=5):
+            audit_log_action: discord.AuditLogAction = audit_log_entry.action
+            if "auto_moderation_user_communication_disabled" in str(audit_log_action):
+                await self._confirm_manual_add_strike(
+                    strike_user=after,
+                    action=discord.AuditLogAction.auto_moderation_user_communication_disabled,
```

Comment:
If action is known to be of correct type, use that action value, rather than re-retrieving that action type.

Fix:
```python
await self._confirm_manual_add_strike(
    strike_user=after,
    action=audit_log_entry.action,
)
```